### PR TITLE
fix(curriculum): customer complaint form edge case

### LINKED ISSUE #59167: Customer Complaint Form - undefined edge case
## Describe the Issue
There's edge case with `#complaint-description` and `#solution-description`, that's not addressed in instructions and test hints.

- If `#other-complaint` is checked and `#complaint-description` contents have less than `20` characters, `validateForm()["solution-description"]` should be `false`.
- If `#other-complaint` is checked and `#complaint-description` contents have at least `20` characters, `validateForm()["solution-description"]` should be `true`.
- It is unclear though what value `validateForm()["solution-description"]` should have, when `#other-complaint` is not checked.
- Similar issue is with the `#other-solution` radio button and `#solution-description`.

Per specification: 

> `validateForm` (...) the value of each key should be `true` if the corresponding form field is correctly filled and `false` otherwise.

> `isValid` that takes the object returned by `validateForm` as argument and returns `true` if every form field is correctly filled and `false` otherwise.

One of the tests is checking if `isValid` returns `true` when `#other-complaint` and `#other-solution` are not checked. This might be the only failing test, when somebody assumes `"solution-description"` or `"complaint-description"` should still validate to `false` if they don't have `20` characters, but their correspondent _Other_ option is not checked. In that case test hint will not help much with narrowing down the issue.
https://github.com/freeCodeCamp/freeCodeCamp/blob/af05efca4a0f697fccce2bfebd8aff1846d4f0fa/curriculum/challenges/english/25-front-end-development/lab-customer-complaint-form/67279fe50237291f80eed8b8.md?plain=1#L311-L330

## Affected Page
https://www.freecodecamp.org/learn/full-stack-developer/lab-customer-complaint-form/build-a-customer-complaint-form

### PR DESCRIPTION
**Pull request by chloeharty**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitHub Codespaces.

Closes #59167 

I added an edge case to fix the customer complaint form. I used a grep search to isolate the issue within the larger curriculum folder. I also tested it manually by running locally on my Windows machine. Please let me know if there is additional work that I need to do.

### CODE REVIEW

File: package.json
Comment (ojeytonwilliams) [inline package.json:1]: Please revert these changes.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

**Comment by chloeharty** [comment]
Hello, I believe that I changed the package.json back to it's original state.

File: curriculum/challenges/english/blocks/lab-customer-complaint-form/67279fe50237291f80eed8b8.md [hunk 1]
```diff
@@ -303,6 +303,13 @@ field.dispatchEvent(new Event("change", { bubbles: true }));
 assert.equal(field.style.borderColor, "green");
 ```
 
+When #other-solution is not checked, validateForm()["solution-description"] should be true.
```
Comment (a2937) [inline curriculum/challenges/english/blocks/lab-customer-complaint-form/67279fe50237291f80eed8b8.md:306]:
```suggestion
When `#other-solution` is not checked, `validateForm()["solution-description"]` should be `true`.
```

**Review by a2937 — CHANGES_REQUESTED** [review]
The problem with your edge-case is that it does not reflect the current behavior of the app. If the `solution-description` is empty, then no value for `solution-description` is returned. And if the `#other-solution` is not checked, the `solution-description` is not able to be edited anyways. 

```js
    if (validationObject["solution-description"] === null) 
    {
        delete validationObject["solution-description"];
    }
```

**Comment by a2937** [comment]
> I have tested these changes either locally on my machine, or GitHub Codespaces.

If you need any help testing the challenge using continuous integration or setting up development on WSL 2 (Windows Subsystem for Linux), do not hesitate to ask either on the forum or on the Discord server.
